pandas/core/indexes/base.py

```diff
@@ -907,16 +907,26 @@ def astype(self, dtype, copy=True):
         """
 
     @Appender(_index_shared_docs["take"] % _index_doc_kwargs)
-    def take(self, indices, axis=0, allow_fill=True, fill_value=None, **kwargs):
+    def take(
+        self,
+        indices: Union[ArrayLike, Sequence[int]],
+        axis: int = 0,
+        allow_fill: bool = True,
+        fill_value=None,
+        **kwargs,
+    ):
```

Comment:
Any advice on typing this return value? Can almost use the TypeVar `_IndexT`, but the problem is that `RangeIndex.take` returns a superclass (`Int64Index`).

Reply:
I _think_ in the base class use `Index`, and can then use a typevar in the subclasses that do return the same type.

Reply:
I tried something along those lines originally - the problem I was running into is how to handle subclasses which rely on parent implementations. For example, for `ExtensionIndex` would like to be able to write a stub like
```
def take(
        self: _ExtensionIndexT,
        indices,
        axis: int = 0,
        allow_fill: bool = True,
        fill_value=None,
        **kwargs,
    ) -> _ExtensionIndexT: ...
```
to essentially type the child while still relying on the inherited implementation. But couldn't figure out a way to achieve this.